### PR TITLE
fix(api-client): too many z-index layers

### PR DESCRIPTION
**Pull request by hanspagel**
I faced a z-index issue in another branch, so I tried to debug it and found a lot of z-index layers throughout the API client. Some in Tailwind, some in CSS, some even conflicting with eachother.

I tried removing all and it didn’t look too bad, so @hwkr added a Tailwind config and reintroduced a z-index where actually needed.

We should test this thoroughly, though. I might have missed places where we need to add it again. Can you all help me with that? 👀 Just leave a screenshot here and we’ll fix it.

### CODE REVIEW

**Comment by antlio** [comment]
great call!

seems like we cannot click the back button anymore
image

**Review by antlio — APPROVED** [review]
beside my comment seems alright! great!

**Review by hwkr — CHANGES_REQUESTED** [review]
Oh yeah we can use tailwind for this, components has a good example of setting [custom z-index config](https://github.com/scalar/scalar/blob/main/packages/components/tailwind.config.ts#L30-L34).

tl;dr;

```js
export default {
  // ...
  theme: {
    extend: {
     // ...,
      zIndex: {
        // Contextual overlays like dropdowns, popovers, tooltips
        context: '1000',
        // Full screen overlays / modals
        overlay: '10000',
      },
    },
  },
}
```

Then you use the class as `z-overlay`.

**Review by hwkr — APPROVED** [review]

**Comment by hwkr** [comment]
Alright I updated this and cleaned up everything that seemed broken, I couldn't figure out how to test the address bar history though. For some reason the button isn't appearing for me even on main 🤔

**Comment by hanspagel** [comment]
In the reference, when I open the API client modal, the background is in front of the modal.

https://976f4431-6bfa-4a93-bb53-eaea404ac630--scalar-deploy-preview.netlify.app/standalone-api-reference

Screenshot 2024-10-08 at 14 15 56

File: packages/api-client/src/layouts/Modal/ApiClientModal.vue [hunk 1]
```diff
@@ -44,12 +44,12 @@ onBeforeUnmount(() => {
...
     v-show="modalState.open"
     class="scalar">
-    <div className="scalar-container">
+    <div className="scalar-container z-overlay">
       <div className="scalar-app scalar-client scalar-app-layout">
         <RouterView key="$route.fullPath" />
       </div>
       <div
-        class="scalar-app-exit"
+        class="scalar-app-exit -z-1"
```
Comment (antlio) [inline packages/api-client/src/layouts/Modal/ApiClientModal.vue:52]: any reason to have the z-overlay here (out of curiosity)? it seems like the `-z-1` is making it

**Comment by hanspagel** [comment]
Awesome work! I think we can merge this, maybe just look into @antlio’s question above. 👌

File: packages/api-client/src/layouts/Modal/ApiClientModal.vue (hunk 1 above)
Comment (hanspagel) [reply packages/api-client/src/layouts/Modal/ApiClientModal.vue:52]: Should we use another Tailwind preset instead of a number here?

Comment (antlio) [reply packages/api-client/src/layouts/Modal/ApiClientModal.vue:52]: can also just add the `z-overlay` as a `scalar-client` class sibling so we don't have to?

Comment (hwkr) [reply packages/api-client/src/layouts/Modal/ApiClientModal.vue:52]: We want to redefined the stacking context for `scalar-app-exit` so it gets moved behind `scalar-client` (which inherently has a `z-index` of 0 in the stacking context) but not behind other elements on the page. 

If we don't make it an overlay there then theres a risk of the app exit getting put behind other high z-index elements on the page the modal is embedded in.

**Comment by hwkr** [comment]
Oops may have found another z-index issue

Google Chrome-2024-10-09-20-29-30@2x
